### PR TITLE
feat: Branding

### LINKED ISSUE #16213: [Task] : Frontend to add admin settings for configuring branding
## Task details

This issue is for frontend team to add brand settings UI in admin settings for EE users to support branding

### LINKED ISSUE #16214: [Task] : Frontend - Changes to login/sign up/password reset to support branding
## Task details
Changes to UI in
1. Login page
2. Sign up page
3. Password reset page 

To support branding and some other css changes

### LINKED ISSUE #16215: [Task] : Frontend - Changes to app dashboard to support branding
## Task details

Changes to the apps dashboard to add brand details for EE users

### LINKED ISSUE #16216: [Task] : Frontend - Changes to apps to support branding
## Task details

Changes in apps in the builder mode and the deployed mode to support branding.

### LINKED ISSUE #16217: [Task] : Frontend - Changes to 404 page to support branding
## Task details

Changes to 404 page to support branding in the EE version and also some css changes in the CE version.

### PR DESCRIPTION
**Pull request by jsartisan**
Allow super admin to set favicon, logo, and primary color according to their brand for their instance. 
image

- updates the look and feel of login/signup pages.
- consumes the values of colors on all CTA
- updates the look and feel of errors pages ( 404, 502, 503, etc )
image

Fixes #16217
Fixes #16216
Fixes #16215
Fixes #16214
Fixes #16213

## Type of change
- New feature

## How Has This Been Tested?
- Manually 
- With Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/ce/sagas/SuperUserSagas.tsx [hunk 1]
```diff
@@ -75,13 +79,22 @@ export function* SaveAdminSettingsSaga(
...
       });
-      yield put({
-        type: ReduxActionTypes.FETCH_ADMIN_SETTINGS_SUCCESS,
-        payload: settings,
-      });
-      yield put({
-        type: ReduxActionTypes.RESTART_SERVER_POLL,
-      });
+
+      if (needsRestart) {
```
Comment (ankitakinger) [inline app/client/src/ce/sagas/SuperUserSagas.tsx:83]: I think the variable name here isn't correct. If I'm not wrong, branding changes after saving don't need a server restart, correct? While other settings need.

File: app/client/src/pages/Settings/FormGroup/ColorInput.tsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  top: 6px;
+  left: 8px;
+  bottom: 0;
+  width: 22px;
+  height: 22px !important;
+  padding-left: 0 !important;
+  border: red;
+  z-index: 1;
+  display: flex;
+  border: 1px solid #e0dede;
```
Comment (ankitakinger) [inline app/client/src/pages/Settings/FormGroup/ColorInput.tsx:67]: we can use CSS variables for color here.

File: app/client/src/ce/sagas/SuperUserSagas.tsx (hunk 1 above)
Comment (jsartisan) [reply app/client/src/ce/sagas/SuperUserSagas.tsx:83]: Oopps. My bad. Let me fix it. I goofed while copying code from EE repo.

File: app/client/src/pages/common/PageHeader.tsx [hunk 3]
```diff
@@ -26,36 +25,23 @@ import MobileSideBar from "./MobileSidebar";
 import { Indices } from "constants/Layers";
 import { Icon, IconSize } from "design-system";
 import { getTemplateNotificationSeenAction } from "actions/templateActions";
+import { getTenantConfig } from "ce/selectors/tenantSelectors";
```
Comment (ankitakinger) [inline app/client/src/pages/common/PageHeader.tsx:28]: use `@appsmit`h here as well in the import

File: app/client/yarn.lock [hunk 4]
```diff
@@ -6236,10 +6236,10 @@ depd@~1.1.2:
   version "1.1.2"
   resolved "https://registry.npmjs.org/depd/-/depd-1.1.2.tgz"
 
-"design-system@npm:@appsmithorg/design-system@1.0.34":
-  version "1.0.34"
-  resolved "https://registry.yarnpkg.com/@appsmithorg/design-system/-/design-system-1.0.34.tgz#ebf63414bec56fecb5c1d5c3464aee91363a151b"
-  integrity sha512-4epvp8Q2X3mH2UV/yvEzdCCrDPiayV2WKR6t6zS5N6O46DIqQfAVbKqANNqEFlMcZMMmzvr6MEBV6PS0arK1yw==
+"design-system@npm:@appsmithorg/design-system@1.0.33-alpha.7":
```
Comment (ankitakinger) [inline app/client/yarn.lock:6239]: I think in `package.json` the version is 37-alpha.7 which doesn't match the version in this file.

File: app/client/src/utils/BrandingUtils.ts [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    });
+
+    return false;
+  }
+
+  // case 4: check size
+  const image = new Image();
+  image.src = window.URL.createObjectURL(file);
+
+  image.onload = function() {
```
[the remark this answers is not in the document]
Comment (jsartisan) [reply app/client/src/utils/BrandingUtils.ts:164]: It's okay. I'll use `.ts` only. `.tsx` was wrong before

File: app/client/yarn.lock (hunk 4 above)
Comment (jsartisan) [reply app/client/yarn.lock:6239]: I'll be using a stable version once i get a sign off from review and qa.

**Review by ankitakinger — COMMENTED** [review]
1. Since you have changed the path for `favicon-orange.ico`, please update the path in `manifest.json` file as well. Also, if it is used in any other files, then those too.
2. Should we add messages to `img` `alt` all over in the new pages you created?

Everything else looks good to me.

File: app/client/public/manifest.json [hunk 6]
```diff
@@ -3,7 +3,7 @@
   "name": "Appsmith Client Web UI",
   "icons": [
     {
-      "src": "favicon-orange.ico",
+      "src": "/static/img/favicon-orange.ico",
```
Comment (ankitakinger) [inline app/client/public/manifest.json:6]: I'm not sure but I think the first `/` is not required. Have you checked this out locally?

**Review by ankitakinger — APPROVED** [review]

**Pull request merged**